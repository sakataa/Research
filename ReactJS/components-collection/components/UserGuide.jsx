import React from 'react';
import ReactDOM from 'react-dom';
import Popup from './popup/Popup';
import PropTypes from 'prop-types';

export default class UserGuide extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            isShow: false
        }
    };

    static propTypes = {
        title: PropTypes.string,
        className: PropTypes.string,
        content: PropTypes.string,
        contentTitle: PropTypes.string
    }

    static defaultProps = {
        className: 'icon-book right',
        content: ''
    }

    onClickHandler = () => {
        this.setState({
            isShow: true
        });

        this.props.onClick && this.props.onClick();
    }

    onCloseHandler = () => {
        this.setState({
            isShow: false
        });
    }

    render() {
        const { className, title, content, contentTitle } = this.props;
        const isShow = this.state.isShow && content.length > 0;
        
        return (<div>
            <i className={className} onClick={this.onClickHandler} title={title}></i>
            <Popup width={'640px'} height={'480px'} isShow={isShow} content={content} title={contentTitle} onClose={this.onCloseHandler} isHtml={true} />
        </div>);
    }
};