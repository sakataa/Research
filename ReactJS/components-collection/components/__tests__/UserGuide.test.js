import React from 'react'
import renderer from 'react-test-renderer'

import UserGuide from '../UserGuide'

describe('UserGuide ', () => {
    describe('render', () => {
        it('popup component is hidden', () => {
            const props = {
                content: '<div></div>'
            };

            const component = renderer.create(<UserGuide {...props} />);

            const result = component.toJSON();
            const popup = result.children[1];
        
            expect(result).toMatchSnapshot();
            expect(popup.props.className).toContain('hide');
        });
    });

    describe('onCloseHandler', () => {
        it('update state to false', () => {
            const component = new UserGuide ({});
            component.setState = jest.fn();

            component.onCloseHandler();

            expect(component.setState).toHaveBeenCalledWith({ isShow: false });
        });
    });

    describe('onClickHandler', () => {
        it('update state to true', () => {
            const component = new UserGuide ({});
            component.setState = jest.fn();

            component.onClickHandler();

            expect(component.setState).toHaveBeenCalledWith({ isShow: true });
        });

        it('onClick prop is called', () => {
            const props = {
                onClick: jest.fn()
            };
            const component = new UserGuide (props);
            component.setState = jest.fn();

            component.onClickHandler();

            expect(props.onClick).toHaveBeenCalled();
        });

        it('update state to true - content is not empty - popup is shown', () => {
            const props = {};

            const component = renderer.create(<UserGuide {...props} />);
            let result = component.toJSON();
            const button = result.children[0];
            const popup = result.children[1];
            button.props.onClick();
            result = component.toJSON()
        
            expect(result).toMatchSnapshot();
            expect(popup.props.className).toContain('hide');
        });

        it('update state to true - content is empty - popup is shown', () => {
            const props = {
                content: '<div></div>'
            };

            const component = renderer.create(<UserGuide {...props} />);
            let result = component.toJSON();
            const button = result.children[0];
            button.props.onClick();
            result = component.toJSON();
            const popup = result.children[1];
        
            expect(result).toMatchSnapshot();
            expect(popup.props.className).toContain('show');
        });
    });
});