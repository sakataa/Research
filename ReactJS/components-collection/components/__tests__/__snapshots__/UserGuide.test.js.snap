// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGuide  onClickHandler update state to true - content is empty - popup is shown 1`] = `
<div>
  <i
    className="icon-book right"
    onClick={[Function]}
    title={undefined}
  />
  <div
    className="popup-container show"
  >
    <div
      className="popup-block-body"
    />
    <div
      className="flat-popup"
      style={
        Object {
          "height": "480px",
          "width": "640px",
        }
      }
    >
      <div
        className="popup-title"
      >
        <div
          className="txt-title"
        />
        <a
          className="popup-close"
          href="javascript:;"
          onClick={[Function]}
        >
          <i
            className="icon-close"
          />
        </a>
      </div>
      <div
        className="popup-content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div></div>",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`UserGuide  onClickHandler update state to true - content is not empty - popup is shown 1`] = `
<div>
  <i
    className="icon-book right"
    onClick={[Function]}
    title={undefined}
  />
  <div
    className="popup-container hide"
  >
    <div
      className="popup-block-body"
    />
    <div
      className="flat-popup"
      style={
        Object {
          "height": "480px",
          "width": "640px",
        }
      }
    >
      <div
        className="popup-title"
      >
        <div
          className="txt-title"
        />
        <a
          className="popup-close"
          href="javascript:;"
          onClick={[Function]}
        >
          <i
            className="icon-close"
          />
        </a>
      </div>
      <div
        className="popup-content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`UserGuide  render popup component is hidden 1`] = `
<div>
  <i
    className="icon-book right"
    onClick={[Function]}
    title={undefined}
  />
  <div
    className="popup-container hide"
  >
    <div
      className="popup-block-body"
    />
    <div
      className="flat-popup"
      style={
        Object {
          "height": "480px",
          "width": "640px",
        }
      }
    >
      <div
        className="popup-title"
      >
        <div
          className="txt-title"
        />
        <a
          className="popup-close"
          href="javascript:;"
          onClick={[Function]}
        >
          <i
            className="icon-close"
          />
        </a>
      </div>
      <div
        className="popup-content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div></div>",
          }
        }
      />
    </div>
  </div>
</div>
`;
